### PR TITLE
Yangfei/constant pad2d

### PR DESCRIPTION
**Pull request by iamyf**
## 概述
实现flow.constant_pad2d，对标pytorch：[torch.nn.ConstantPad2d](https://pytorch.org/docs/stable/generated/torch.nn.ConstantPad2d.html)
- 支持 NCHW 格式输入 
- 支持输入 int, list, tuple 类型的 padding 参数
### Op: constant_pad2d_op
 - [x] Op SetBatchAxisInferFn
 - [x] Op SetGetSbpFn
 - [x] Op SetInputArgModifyFn
 - [x] Op 反向梯度注册

### Kernel: constant_pad2d_kernel
 - [x] CPU in:float32 
 - [x] CPU in:float64
 - [x] CPU in:int32
 - [ ] CPU in:int64
 - [ ] CPU in:int8

 - [x] GPU in:float32 
 - [x] GPU in:float64
 - [x] GPU in:int32
 - [ ] GPU in:int64
 - [x] GPU in:float16
 - [ ] GPU in:int8

### Python Wrapper
 - [x] Python API 参数检查及异常提示
 - [x] 接口注释
 - [x] Example 

### 测试
 - [x] 单机单卡  CPU Test Case 
 - [x] 单机单卡  GPU Test Case 
 - [x] 单机多卡  CPU Test Case 
 - [x] 单机多卡  GPU Test Case
 - [ ] 分布式  CPU Test Case  
 - [ ] 分布式  GPU Test Case  

## GPU 有效带宽
理论带宽：
```text
 Device to Device Bandwidth, 1 Device(s)
 PINNED Memory Transfers
   Transfer Size (Bytes)        Bandwidth(GB/s)
   32000000                     729.7
```

实际带宽：
```
PROFILER::KERNEL::CUDA_MEMORY_BANDWIDTH op_name: Constant_Pad2d2 elapsed(ms): 0.117728 memory_size(Byte): 50430048 bandwidth(GB/s): 398.942
PROFILER::KERNEL::CUDA_MEMORY_BANDWIDTH op_name: Constant_Pad2d2_grad elapsed(ms): 0.153632 memory_size(Byte): 50430048 bandwidth(GB/s): 305.709
```

## PR Checklist
 - [x] PR 标题语句通畅，明确表达 PR 内容，适合直接作为新版本发布时的 changelog
 - [x] 代码格式化 
 - [x] 已经本地编译通过
 - [x] 已本地针对改动测试
 - [x] 已添加 type 标签:(填写 type 标签名，如 `bug, enhancement, purge, feature, documentation`)
 - [x] 已添加 component 标签:(填写 component 标签名，如 `op, system, eager, build, xla, python, ci, test, tooling, onnx`)
 - [x] Draft 转正式 PR 前已请人 Review

### CODE REVIEW

File: oneflow/python/test/ops/test_constant_pad2d.py [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+import os
+from collections import OrderedDict
+
+import numpy as np
+import oneflow as flow
+import oneflow.typing as tp
+from test_util import Args, GenArgDict, GenArgList
+
+
+def flatten_array(input_array):
```
Comment (Flowingsun007) [inline oneflow/python/test/ops/test_constant_pad2d.py:26]: numpy相关的生成测试矩阵、求梯度还有像flatten_array、array_to_numpy 、index2coordinate 、coordinate2index这些公共方法可以提取到util工具类里，方便以后复用

Comment (iamyf) [reply oneflow/python/test/ops/test_constant_pad2d.py]: 这几个方法添加到test_util 里了

**Review by Flowingsun007 — APPROVED** [review]
